src/g_editor.c

```diff
@@ -2580,6 +2651,69 @@ static void canvas_doclick(t_canvas *x, int xpos, int ypos, int which,
     }
 }
 
+#ifdef HAVE_KEYBOARDNAV
+void canvas_goto(t_canvas *x, t_floatarg indexarg)
+{
+    int index = indexarg;
+    if ( index < 0)
+    {
+        post("go to error (negative index): %d", index);
+        return;
+    }
+    t_gobj *gobj = x->gl_list;
+    if ( !gobj )
+    {
+        error("canvas_goto() error: patch is empty.");
+        return;
+    }
+    glist_noselect(x);
+    int count = 0;
+    while ( gobj->g_next )
+    {
+        if ( count == index )
+            break;
+        gobj = gobj->g_next;
+        ++count;
+    }
+    if ( count < index )
+    {
+        post("go to error (no such element): %d. Patch only has %d elements", index,count);
+        return;
+    }
+    canvas_redraw(x);
+    if( !glist_isselected(x, gobj) )
+    {
+        t_editor *editor = x->gl_editor;
+        if(editor)
+        {
+            t_editor_private *private = editor->e_privatedata;
+            t_kbdnav *kbdnav = &private->kbdnav;
```

Comment:
*always use `canvas_get_kbdnav()` instead of directly accessing the è_privatedata`.
(there are multiple occurences of this anti-pattern in the code)